keras/engine/training.py

```diff
@@ -3879,10 +3879,15 @@ def reduce_per_replica(values, strategy, reduction):
 
     def _reduce(v):
         """Reduce a single `PerReplica` object."""
-        if reduction == "concat" and _collective_all_reduce_multi_worker(
-            strategy
-        ):
-            return _multi_worker_concat(v, strategy)
+        if reduction in (
```

Comment:
Suggestion: do you think this can work?

```
if _collective_all_reduce_multi_worker(strategy):
        if reduction == "concat":
            return _multi_worker_concat(v, strategy)
        elif reduction == "sum":
            return strategy.reduce("SUM", v, axis=None)
```

This appears slightly simplified to me